js/src/common/components/ModalManager.js

```diff
@@ -46,10 +50,23 @@ export default class ModalManager extends Component {
     this.$()
       .one('shown.bs.modal', readyCallback)
       .modal({
-        backdrop: dismissible || 'static',
-        keyboard: dismissible,
+        // 'static' means that a backdrop click doesn't dismiss the modal
+        backdrop: backdropDismissible ? true : 'static',
+        keyboard: keyboardDismissible,
       })
       .modal('show');
+
+    // Disabling backdrop dismissal also disables keyboard dismissibility
+    // but we want to be able to provide key-based dismissal while having
+    // backdrop-based disabled. To do this, we set up our own event
+    // handler to hide the modal if we press Escape.
+    if (!backdropDismissible && keyboardDismissible) {
+      this.$().on('keypress', function (e) {
+        if (e.key === 'Escape') {
+          this.animateHide();
```

Comment:
Not a change request but a comment for reviewers: we'll need to test that dismissing this way doesn't result in weird UI glitches.